### PR TITLE
ProductList filter functionality

### PR DESCRIPTION
**Pull request by varun-Tad**
I have added filter functionality for ProductList. Please review it.You can navigate to Product list page by clicking on Shop now button

Files to review:
1. reducerFn.js
2. FilterFunctions.js
3. Product-context.js
4. ProdListingPage.js

Preview: https://visionary-flan-13c07d.netlify.app/

### CODE REVIEW

File: dressup/src/pages/ProductsPage/FilterFunctions.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    case "highToLow":
+      return products.sort(
+        (a, b) => b.price.discountPrice - a.price.discountPrice
+      );
+    case "lowToHigh":
+      return products.sort(
+        (a, b) => a.price.discountPrice - b.price.discountPrice
+      );
+    default:
+      return products;
```
Comment (rdthetha) [inline dressup/src/pages/ProductsPage/FilterFunctions.js:14]: .sort() mutates the original array. Either make a copy of the array or use spread operator!

File: dressup/src/pages/ProductsPage/FilterSection/FilterSection.js [hunk 2]
```diff
@@ -27,6 +32,9 @@ const FilterSection = () => {
               type="radio"
               name="sort-by"
               value="Price low to high"
+              onChange={() =>
+                dispatch({ type: "lowToHigh", value: "lowToHigh" })
+              }
             ></input>
             <label htmlFor="Price low to high">Price low to high</label>
```
Comment (sri-rishi) [inline dressup/src/pages/ProductsPage/FilterSection/FilterSection.js:39]: Suggestion: provide `id` attribute with same value as `htmlFor` for connecting them

File: dressup/src/pages/ProductsPage/FilterSection/FilterSection.js [hunk 3]
```diff
@@ -46,51 +57,101 @@ const FilterSection = () => {
...
           <p>CATEGORIES</p>
           <li>
-            <input type="checkbox" name="categories" value="Tshirts"></input>
-            <label htmlFor="Tshirts"> Tshirts</label>
+            <input
+              type="checkbox"
+              name="Pants"
+              value="Pants"
+              checked={state.categories.Pants}
+              onChange={(e) => dispatch({ type: "catg", value: e.target.name })}
```
Comment (sri-rishi) [inline dressup/src/pages/ProductsPage/FilterSection/FilterSection.js:65]: Suggestion: use Descriptive naming, it's to have long name when they are making sense

File: dressup/src/pages/ProductsPage/ProdListingPage/ProdListingPage.js [hunk 4]
```diff
@@ -2,11 +2,30 @@ import "./ProdListingPage.css";
...
 const ProdListingPage = () => {
+  const { state } = useProduct();
+  console.log("state", state);
+  const sortedProducts = sortProductList(state.sortOrder, ProductsPageData);
+
+  const ratedProducts = sortRatingList(state.rating, sortedProducts);
+
+  const checkBoxProducts = sortingCatgeories(state.categories, ratedProducts);
+
+  const RangedProducts = sortPriceRange(state.theValue, checkBoxProducts);
```
Comment (sri-rishi) [inline dressup/src/pages/ProductsPage/ProdListingPage/ProdListingPage.js:23]: Suggestion: These can be extract in separated files and used here

File: dressup/src/pages/ProductsPage/ProductsPage.data.js [hunk 5]
```diff
@@ -2,11 +2,16 @@ import img1 from "../../images/jeans.jpeg";
 import img2 from "../../images/denim-jeans.jpeg";
 import img3 from "../../images/top.jpeg";
 import img4 from "../../images/jeans-two.jpeg";
+import img5 from "../../images/women-watch.jpeg";
+import img6 from "../../images/mens-watchOne.jpeg";
+import img7 from "../../images/MensWatch.webp";
+import img8 from "../../images/MensShirtTwo.webp";
+import img9 from "../../images/MensShirtThree.webp";
```
Comment (sri-rishi) [inline dressup/src/pages/ProductsPage/ProductsPage.data.js:10]: Suggestion: you can import images in one file nd then imported from there, that will make file clean

File: dressup/src/pages/ProductsPage/ProdListingPage/reducerFn.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+      return { ...state, rating: action.value };
+    case "FourStars":
+      return { ...state, rating: action.value };
+    case "ThreeStars":
+      return { ...state, rating: action.value };
+    case "TwoStars":
+      return { ...state, rating: action.value };
+    case "OneStar":
+      return { ...state, rating: action.value };
+    case "All":
```
Comment (sri-rishi) [inline dressup/src/pages/ProductsPage/ProdListingPage/reducerFn.js:17]: Suggestion: you can here give one type for rating like "FILTER_BY_RATINGS` and can sort by `action.value`

**Review by Sahil25061999 — COMMENTED** [review]
same suggestion as mentioned above, also try to display something like "No data available" if the there are no products matching the filters, also the "clear all" functionality is not working.

**Pull request merged**